lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb

```diff
@@ -29,18 +29,18 @@ def initialize(sandbox, project, target, umbrella_headers_by_dir = nil)
           #
           def install!
             UI.message "- Installing target `#{target.name}` #{target.platform}" do
+              create_support_files_dir
```

Comment:
Reminder to myself to update #7845 to include this scenario if necessary - there I changed it so that we clean at the end of install, but I might need to add a call to clean within the `unless target.should_build?` section

Reply:
Whoops, totally misread this, thought it was manually creating the temporary directory.